dodo/search.py

```diff
@@ -64,20 +74,34 @@ def refresh_thread(self, thread: QModelIndex|str):
             row = thread.row()
             assert thread_id is not None
 
-        r = subprocess.run(['notmuch', 'search', '--format=json', f'{self.q} AND thread:{thread_id}'],
-                stdout=subprocess.PIPE)
-        contents = json.loads(r.stdout.decode('utf-8'))
-
+        logger.info("Search '%s': refreshing thread %s", self.q, thread_id)
         self.beginResetModel()
-        self.d[row:row+1] = contents
-        self.threads = {thread['thread']: i for i,thread in enumerate(self.d)}
+        try:
+            r = subprocess.run(['notmuch', 'search', '--format=json', f'{self.q} AND thread:{thread_id}'],
+                    stdout=subprocess.PIPE)
```

Comment:
Needs:

- `capture_output=True` to be able to access `e.stderr` below
- `check=True` to actually get a `CalledProcessError` if this fails
- Possibly `text=True` for the same reasons as above

```suggestion
                    capture_output=True, text=True, check=True)
```

for the same reasons as above.

Reply:
Fixed, thanks!